pkg/operator/ceph/csi/config.go

```diff
@@ -178,3 +154,34 @@ func createUpdateClientProfile(c client.Client, clusterInfo *cephclient.ClusterI
 
 	return nil
 }
+
+func applyCephFSMountOptions(clusterInfo *cephclient.ClusterInfo, cephFs *csiopv1.CephFsConfigSpec) {
+	if reflect.DeepEqual(clusterInfo.CSIDriverSpec.CephFS, cephv1.CSICephFSSpec{}) {
```

Comment:
Do we really need this check? Seems like if it's the empty struct, then anyway the conditions in the rest of this method will just be skipped anyway.
```suggestion
```

Reply:
@travisn we need to validate `cephv1.CSICephFSSpec` is not nil as the rest of the method call `.CephFS.KernelMountOptions` which is under `cephv1.CSICephFSSpec` so if `cephv1.CSICephFSSpec` is nil it will cause error. And `cephv1.CSICephFSSpec` is pointer struct that's why added this check also if you check the previous code [here](https://github.com/rook/rook/pull/16837/files#diff-b4152c743036caddafbf4e035718164f8a6795afb7d17c1813285d80aabddcf9L113) we added this.